test/e2e_tsa_mtls.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+#
+# Copyright 2023 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -exuo pipefail
+
+## Requirements
+# - cosign
+# - crane
+# - go
+
+CERT_BASE="test/testdata"
+
+TIMESTAMP_CACERT=$CERT_BASE/tsa-mtls-cacert.pem
+TIMESTAMP_CLIENT_CERT=$CERT_BASE/tsa-mtls-client.pem
+TIMESTAMP_CLIENT_KEY=$CERT_BASE/tsa-mtls-client-key.pem
+TIMESTAMP_SERVER_CERT=$CERT_BASE/tsa-mtls-server.pem
+TIMESTAMP_SERVER_KEY=$CERT_BASE/tsa-mtls-server-key.pem
+TIMESTAMP_SERVER_NAME="server.example.com"
+TIMESTAMP_SERVER_URL=https://localhost:3000/api/v1/timestamp
+
+rm -fr /tmp/timestamp-authority
+git clone https://github.com/sigstore/timestamp-authority /tmp/timestamp-authority
+pushd /tmp/timestamp-authority
+make
+popd
+/tmp/timestamp-authority/bin/timestamp-server serve --disable-ntp-monitoring --tls-host 0.0.0.0 --tls-port 3000 \
+	--scheme https --tls-ca $TIMESTAMP_CACERT --tls-key $TIMESTAMP_SERVER_KEY --tls-certificate $TIMESTAMP_SERVER_CERT &
+export PATH="/tmp/timestampserver:$PATH"
+
+IMG=${IMAGE_URI_DIGEST:-}
+if [[ "$#" -ge 1 ]]; then
+	IMG=$1
+elif [[ -z "${IMG}" ]]; then
+	# Upload an image to ttl.sh - commands from https://docs.sigstore.dev/cosign/keyless/
+	SRC_IMAGE=busybox
+	SRC_DIGEST=$(crane digest busybox)
+	IMAGE_URI=ttl.sh/$(uuidgen | head -c 8 | tr 'A-Z' 'a-z')
+	crane cp $SRC_IMAGE@$SRC_DIGEST $IMAGE_URI:3h
+	IMG=$IMAGE_URI@$SRC_DIGEST
+fi
+
+echo "IMG (IMAGE_URI_DIGEST): $IMG, TIMESTAMP_SERVER_URL: $TIMESTAMP_SERVER_URL"
+
+GOBIN=/tmp GOPROXY=https://proxy.golang.org,direct go install -v github.com/dmitris/gencert@latest
```

Comment:
Instead of using the online CA in [certificates](https://github.com/smallstep/certificates), a couple of invocations of [step](https://github.com/smallstep/cli) in the `--offline` [mode](https://smallstep.com/docs/step-ca/certificate-authority-core-concepts/#offline-mode) might also work for this test. Haven't looked through the code yet, though, so I may be wrong.

There's also a GitHub action to install the CLI: https://github.com/smallstep/setup-step-cli. That might also be useful.

Reply:
for now, I checked in `test/gencert/main.go` and used it in the `e2e_tsa_mtls.sh`, getting rid of cloning my personal repo.  Could we start with that, and then convert it to using smallstep's `step`?

Reply:
Sounds good!